orchagent/portsorch.cpp

```diff
@@ -9024,3 +9043,55 @@ void PortsOrch::doTask(swss::SelectableTimer &timer)
     }
 }
 
+void PortsOrch::updateVoqStatsPerPortChassisDb(const Port& port)
+{
+    std::vector<sai_object_id_t> queue_ids;
+    static const vector<sai_stat_id_t> queueStatIds =
+    {
+        SAI_QUEUE_STAT_PACKETS,
+        SAI_QUEUE_STAT_BYTES,
+        SAI_QUEUE_STAT_DROPPED_PACKETS,
+        SAI_QUEUE_STAT_DROPPED_BYTES,
+    };
+    vector<uint64_t> queueStats(queueStatIds.size());
+
+    queue_ids = m_port_voq_ids[port.m_alias];
+    SWSS_LOG_ENTER();
+
+    for (size_t queueIndex = 0; queueIndex < queue_ids.size(); ++queueIndex)
+    {
+        std::ostringstream key;
+        key << port.m_system_port_info.alias << "@" << gMyHostName << "|" << gMyAsicName << ":" << queueIndex;
+
+        sai_status_t status = sai_queue_api->get_queue_stats(
+           queue_ids[queueIndex],
+           static_cast<uint32_t>(queueStatIds.size()),
+           queueStatIds.data(),
+           queueStats.data());
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+           SWSS_LOG_ERROR("Failed to fetch queue 0x%" PRIx64 " status: %d", queue_ids[queueIndex], status);
+           continue;
+        }
+
+        //Add to CHASSIS_APP_DB
+        int index = 0;
+        for (const auto& it: queue_stat_ids)
+        {
+            SWSS_LOG_DEBUG("%s %s %s",key.str().c_str(), sai_serialize_queue_stat(it).c_str(),
+                           to_string(queueStats[index]).c_str());
+            m_tableVoqQueueCounter->hset(key.str(), sai_serialize_queue_stat(it), to_string(queueStats[index++]));
```

Comment:
1. Is this feature supported on SUP ? 
2. Please share CLI command as well as o/p.
3. Please explain how aggregation of VoQ Stats from multiple LCs happening here. As VoQ id may be same across multiple LCs, it may end up over-writing existing data.